examples/http/src/main/java/io/opentelemetry/example/http/HttpClient.java

```diff
@@ -53,13 +53,19 @@ private void makeRequest() throws IOException {
       span.setAttribute(SemanticAttributes.HTTP_METHOD, "GET");
       span.setAttribute("component", "http");
       /*
-       Only one of the following is required:
+       Only one of the following is required
          - http.url
          - http.scheme, http.host, http.target
          - http.scheme, peer.hostname, peer.port, http.target
          - http.scheme, peer.ip, peer.port, http.target
       */
-      span.setAttribute(SemanticAttributes.HTTP_URL, url.toString());
+
+      if (url.getUserInfo() == null || url.getUserInfo().isEmpty()) {
+        span.setAttribute(SemanticAttributes.HTTP_URL, url.toString());
+      } else {
+        span.setAttribute(
+            SemanticAttributes.HTTP_URL, url.toString().replace(url.getUserInfo() + '@', ""));
```

Comment:
In the spirit of code safety, I'd apply here same logic as suggested for changes in the instrumentation module: 

```
new URI(
    url.getScheme(),
    null,
    url.getHost(),
    url.getPort(),
    url.getPath(),
    url.getQuery(),
    url.getFragment()).toUrl();
```

Reply:
Yes, this is definitely a safer way to do this work. @Hangzhi Can you make this change as well? With this, you can also get rid of the if/else and just do it once using the URI to remove the user info and format correctly.